retrieval/transform_test.go

```diff
@@ -760,6 +761,121 @@ func TestSampleBuilder(t *testing.T) {
 				},
 			},
 		},
+		// Samples with a NaN value should be dropped.
+		{
+			targets: targetMap{
+				"job1/instance1": &targets.Target{
+					Labels:           promlabels.FromStrings("job", "job1", "instance", "instance1"),
+					DiscoveredLabels: promlabels.FromStrings("__resource_a", "resource2_a"),
+				},
+			},
+			series: seriesMap{
+				1: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1_count"),
+			},
+			metadata: metadataMap{
+				"job1/instance1/metric1": &scrape.MetricMetadata{Type: textparse.MetricTypeSummary, Metric: "metric1_count"},
+			},
+			metricPrefix: "test.googleapis.com",
+			input: []tsdb.RefSample{
+				// Need multiple samples, since the first will always be nil due to reset timestamp handling
+				{Ref: 1, T: 2000, V: 5},
```

Comment:
Nit: As we are testing the `NaN` mainly, I think using 2 tsdb.RefSample as 
```
[]tsdb.RefSample{
				{Ref: 1, T: 2000, V: 5},
                                {Ref: 1, T: 4000, V: math.NaN()},
}
```
will help the reader understand this testcase more easily.

What do you think? If you agree, do not forget to apply this to the testcase next to this one.

Reply:
Agreed.